src/include/sof/audio/module_adapter/intel/syntax_sugar.h

```diff
@@ -0,0 +1,248 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ */
+
+#ifndef SYNTAX_SUGAR_H
+#define SYNTAX_SUGAR_H
+
+/*
+ * Various helper macros which have primary role of making code more readable.
+ * Do not encode anything platform specific here
+ */
+/*! Given IP is present in HW */
+#define PRESENT(feature) CONFIG_ADSP_##feature##_PRESENT
+/*! Given IP is NOT in HW */
+#define NOT_PRESENT(feature) (CONFIG_ADSP_##feature##_PRESENT == 0)
+#define SUPPORTED(feature) CONFIG_ADSP_##feature##_SUPPORT
+#define NOT_SUPPORTED(feature) (CONFIG_ADSP_##feature##_SUPPORT == 0)
+#define REQUIRED(feature) CONFIG_ADSP_##feature##_REQUIRED
+#define NOT_REQUIRED(feature) (CONFIG_ADSP_##feature##_REQUIRED == 0)
+#define WORKAROUND(feature) CONFIG_ADSP_##feature##_WORKAROUND
+#define NOT_WORKAROUND(feature) (CONFIG_ADSP_##feature##_WORKAROUND == 0)
+#define GET_QUANTITY(feature) CONFIG_ADSP_##feature##_QUANTITY
+#define GET_SIZE(feature) CONFIG_ADSP_##feature##_SIZE
+#define GET_VALUE(feature) CONFIG_ADSP_##feature##_VALUE
+#define ADDRESS(feature) CONFIG_ADSP_##feature##_ADDRESS
+#define REGISTER(feature) CONFIG_ADSP_##feature##_REGISTER
+#define REGISTER_CSI(base, feature) REGISTER(feature)(base)
+#define LOCATION(feature) CONFIG_ADSP_##feature##_LOCATION
+#define VERSION(feature) CONFIG_ADSP_##feature##_VERSION
+#define REVISION(feature) CONFIG_ADSP_##feature##_REVISION
+#define REGISTER_PRESENT(reg_name) CONFIG_ADSP_##reg_name##_REGISTER_PRESENT
+#define GET_UUID(feature) CONFIG_ADSP_##feature##_UUID
```

Comment:
Some of these are very generic and may collide with top level macros e.g. VERSION. Whats the plan to make sure we dont collide ? a macro prefix would work here.

Reply:
syntax_sugar.h removed since needed only for IADK module build. No needed for adapter.